Allow building with containers

riyad opened the pull request:
Maybe you're interested in also allowing/simplifying building with containers. :smile:

riyad committed:
Allow building with containers

richardelling commented:
I'm not opposed to building in a container, but the container and its ZFS version needs to match almost exactly to the deployment environment. So while you've nicely supplied an ubuntu:rolling environment, the resulting binary could be useless to someone on CentOS. There really isn't an easier answer than to build in the same environment as deployment.

riyad commented:
Hmm ... you're right. Maybe the file could be named something like `Dockerfile.ubuntu`.

FWIW: In my limited testing a binary built on the latest Ubuntu release (i.e. 21.04) worked at least on the latest LTS (20.04) ... so there seems to be at least some intra-family backwards compatibility.

riyad commented:
I've renamed the Dockerfile to signify that it's distro-specific. I've also added a paragraph to the README to drive this point home.

FYI: I've also rebased the PR.
